config_test.go

```diff
@@ -0,0 +1,123 @@
+	// [/tmp/go-build689616692/b001/dcrdata.test -test.v=true
+	// -test.run=TestDefaultConfigAPIListen]
+	// however, the go-flags package then parses these and fails
+	// with unknown flags
+	// you will need to remove these flags before the flag parser is
+	// called like: os.Args = os.Args[:1]
+	os.Args = os.Args[:1]
+}
+
+func TestDefaultConfig(t *testing.T) {
+	c := defaultConfig
+	if c.ConfigFile != defaultConfigFile {
+		t.Errorf("expected %s config %s ", c.ConfigFile, defaultConfigFile)
+	}
+}
+
+func TestDefaultConfigFile(t *testing.T) {
+	u, _ := user.Current()
+	dcrpath := filepath.Join(u.HomeDir, ".dcrdata", "dcrdata.conf")
+	if defaultConfigFile != dcrpath {
+		t.Errorf("expected %s config %s ", defaultConfigFile, dcrpath)
+		t.Errorf("Failed to load default dcrdata config")
+	}
+}
+
+func TestLoadConfig(t *testing.T) {
+	_, err := loadConfig()
+	if err != nil {
+		t.Errorf("Failed to load dcrdata config: %s\n", err.Error())
+	}
+}
+
+func TestDefaultConfigAPIListen(t *testing.T) {
+	cfg, err := loadConfig()
+	if err != nil {
+		t.Errorf("Failed to load dcrdata config: %s\n", err.Error())
+	} else {
+		v := cfg.APIListen
+		if v != "127.0.0.1:7777" {
+			t.Errorf("config expected %s but returned %s", "127.0.0.1:7777", v)
+		}
+	}
+
+}
+
+func TestDefaultConfigAPIListenWithEnv(t *testing.T) {
+	os.Setenv("LISTEN_URL", "0.0.0.0:7777")
+	cfg, _ := loadConfig()
+	v := cfg.APIListen
+	if v != "0.0.0.0:7777" {
+		t.Errorf("config expected %s but returned %s", "0.0.0.0:7777", v)
+	}
+}
+
+func TestDefaultConfigHomeDir(t *testing.T) {
+	u, _ := user.Current()
+	cfg, _ := loadConfig()
+	v := cfg.HomeDir
+	expected := filepath.Join(u.HomeDir, ".dcrdata")
+	if v != expected {
+		t.Errorf("config expected %s but returned %s", expected, v)
+	}
+}
+
+func TestDefaultConfigHomeDirWithEnv(t *testing.T) {
+	os.Setenv("HOME_DIR", "/tmp/test")
+	cfg, _ := loadConfig()
+	v := cfg.HomeDir
+	expected := filepath.Join("/tmp/test")
+	if v != expected {
+		t.Errorf("config expected %s but returned %s", expected, v)
+	}
+}
+
+// test to ensure that the flags override env variables
+func TestDefaultConfigHomeDirWithEnvAndFlag(t *testing.T) {
+	os.Args = append(os.Args, "--appdata=/tmp/test2")
+	os.Setenv("HOME_DIR", "/tmp/test")
+	cfg, _ := loadConfig()
+	v := cfg.HomeDir
+	expected := filepath.Join("/tmp/test2")
+	if v != expected {
+		t.Errorf("config expected %s but returned %s", expected, v)
+	}
+}
+
+func TestDefaultConfigTestNet(t *testing.T) {
+	cfg, _ := loadConfig()
+	v := cfg.TestNet
+	if v {
+		t.Errorf("config expected true but returned %+v", v)
+	}
+}
+
+func TestDefaultConfigTestNetWithEnv(t *testing.T) {
+	os.Setenv("USE_TESTNET", "true")
+	cfg, _ := loadConfig()
+	v := cfg.TestNet
+	if !v {
+		t.Errorf("config expected true but returned %+v", v)
+	}
+}
+
+func TestDefaultConfigTestNetWithEnvAndBadValue(t *testing.T) {
+	os.Setenv("USE_TESTNET", "no")
+	cfg, _ := loadConfig()
+	v := cfg.TestNet
+	// should show error and continue
```

Comment:
I would expect a parse error would be an error, and that you could simply return `err` from `loadConfig` and check that.